EventsExpress/Mapping/EventMapperProfile.cs

```diff
@@ -104,49 +98,81 @@ public EventMapperProfile()
                 .ForMember(dest => dest.Visitors, opts => opts.Ignore());
         }
 
+        private static LocationDto MapLocation(Event e)
+        {
+            if (e.EventLocation != null)
+            {
+                return new LocationDto()
+                {
+                    Id = e.Id,
+                    OnlineMeeting = e.EventLocation.OnlineMeeting,
+                    Point = e.EventLocation.Point,
+                    Type = e.EventLocation.Type,
+                };
+            }
+
+            return null;
+        }
+
+        private static LocationDto MapLocation(EventViewModelBase e)
+        {
+            LocationDto locationDto = null;
+
+            if (e.Location != null)
+            {
+                locationDto = new LocationDto();
+                if (e.Location.Type != null)
+                {
+                    locationDto.Type = e.Location.Type;
+                }
+
+                if (e.Location.OnlineMeeting != null && e.Location.Type == LocationType.Online)
+                {
+                    locationDto.OnlineMeeting = e.Location.OnlineMeeting;
+                }
+
+                if (e.Location.Type == LocationType.Map)
+                {
+                    locationDto.Point = e.GetType() == typeof(EventCreateViewModel) ?
```

Comment:
Why do we need this check for `EventCreateViewModel` here?

Reply:
This check is needed because we have to know what method of point creation to invoke. [PointOrNullCreate] is supposed to be invoked on EventCreate, but [PointOrNullEdit] on EventEdit.

Reply:
resolved